### DISCUSSION TITLE
Finding Device with EbusPicLoader

### DISCUSSION THREAD

**Discussion by Bodo1706** [john30/ebusd#792] [Q&A]
Hi All,
i accidentially saved a wrong config during after logging to the adapter in AP mode. Therefore I set my EBus WIFI 3.1 Adapter into bootloading mode. Now is try to configure it again.

I downloaded the EBUSPICLOADER to my Windows PC, connected the adapter to an USB port and now I don't now how to adress the USB device.

ebuspicloader --variant=W COM1 leads to the error "unable to open COM1"

I dont now how to adress the USB port in Cygwin nor cmd.exe.

THX for helping.

BR
Bodo

**Comment by andrewt27** [comment]
Is there any other way to upload this firmware? I have a mac, I tried to do it on a laptop (Windows) and it didn't work. I can't dress USB port. I installed CygWin and also some errors with libraries popped up.

**Reply by Bodo1706** [reply]
As I understood so far, you have to distnguish between uploading the firmware and setting the board to wireless mode.
In Cygwin, Windows COM ports are mapped under /dev/ttyS*, e.g. COM1 would be /dev/ttyS0.
For me it helped not to install Cygwin (the cywin files when you download epicpicloader for windos are sufficient. You can do it with a cmd window as long as the cygwin files are in the same directory. (see https://github.com/john30/ebusd/blob/master/src/tools/README.md). 
Short sequence: 1) unplug Wifi modul. 2) Connect via USB to PC 3) Setting it to Boatloader mode by bridging Pins 3 and 4 of J11. 4) using CMD window to do what you want.

Btw: I also hade some libary errors with different versions of cygwin files. I think it should not happen if you only use the cygwin files of the ebuspicloader delivery in cmd window.

Hope it helps
Bodo

**Reply by andrewt27** [reply]
Hi, thanks for your reply.
I did exactly as it is in the manual and as you described. I removed Cygwin also. I tried make it on two windows laptops and nothing. I still keep getting the error "unable to open /dev/ttyS*" or "unable to open /dev/ttyUSB*" (where * is number from 0 to 10).
Gosh, how frustrating it is. I've wasted hours and no result :(

**Reply by Bodo1706** [reply]
Hi, 
are you aware that ttySX = COM(X+1)? That means for COM6 you have to adress ttyS5.
Greets Bodo

**Reply by john30** [reply]
> Hi, thanks for your reply. I did exactly as it is in the manual and as you described. I removed Cygwin also. I tried make it on two windows laptops and nothing. I still keep getting the error "unable to open /dev/ttyS*" or "unable to open /dev/ttyUSB*" (where * is number from 0 to 10). Gosh, how frustrating it is. I've wasted hours and no result :(

on Windows, check if a COM port appears in the device manager when you connect the adapter (without anything else as documented i.e. without Wemos or USR-ES1 module). Once you have that COM port X, you can connect as /dev/ttySY with Y being X-1.
Alternatively, let ebuspicloader find the device for you using "/dev/ttyS*"

**Reply by karusell** [reply]
Hello @andrewt27,

have same issue with this. 

I get "unable to open /dev/ttyS*". I tried in CMD and Cygwin on two different PCs.
@andrewt27 How have you solved this issue in the end? would be great to know.

Question @ all  : Is there a specific driver for USB required?

@john30 When I let ebuspicloader find the device with "/dev/ttyS*" the response in the cmd is "unable to open /dev/ttyS3". Does that mean the right port could be ttyS3 (COM2) but the permission is not given or the bootloader (J11- Pins3 and 4 brigded for short time and the blue like accures) hasn't been started right?

This is how it looks if I started bootloader. I tried to use the brigde shortly only to get in the blue light and when it didn't work to open I tried in other case to let the bridge there. Both didn't work (see issue on top).

![blue light](https://github.com/john30/ebusd/assets/43848891/51c22af6-b734-4a3e-8607-8b56c2935333)

Thanks for your help. 
Like really to solve this and change my eBUS Adpater V3.1 factory setting "USB MODE" to "Ethernet MODE" via ebuspicloader. #1116

**Reply by cybersmart-eu** [reply]
I solved it yesterday with my adapter v3.1 and also flashed the latest firmware successfully.
There is several potential issues I experienced.
1) USB cable - some cables seem to be power only, thus the PC does not detect the device. Check in device manager and ensure you have a cable where the interface comes up

2) Windows has a driver available, however you need to connect the device so that it pops up in device manager (typically with a ! as no driver is installed yet). Next go to Windows Update, run an update and choose the optional driver update for the device which will be offered there. After that it will be properly connected (in my case it was on COM7)

3) bootloader mode needs to be enabled to flash firmware/configure the interface, but acc. to your pic with the yellow cables you know what to do

I had to try several cables as some did not allow flashing without errors - 🙅 ng during flash or reporting wrong checksum at the end.

Good Luck and a 🥳 happy new year
